recipes/ohnet/all/conanfile.py

```diff
@@ -0,0 +1,149 @@
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def generate(self):
+        if is_msvc(self):
+            tc = NMakeToolchain(self)
+            tc.generate()
+        else:
+            tc = AutotoolsToolchain(self)
+            tc.make_args.append("-j1")
+            tc.generate()
+
+    def build(self):
+        apply_conandata_patches(self)
+        targets = "ohNetDll TestsNative proxies devices"
+        additional_options=""
+
+        with chdir(self, self.source_folder):
+            if is_msvc(self):
+                if self.settings.build_type == "Debug":
+                    additional_options += " debug=1"
+                self.run(f"nmake /f OhNet.mak {targets} {additional_options}")
+            else:
+                args = []
+                args = self._get_openhome_architecture(args)
+                args.append("rsync=no")
+                if str(self.settings.compiler.libcxx) == "libc++":
+                    args.extend(["CPPFLAGS=-stdlib=libc++", "LDFLAGS=-stdlib=libc++"])
+                autotools = Autotools(self)
+                autotools.make(args=args, target=targets)
+
+    def package(self):
+        installlibdir = os.path.join(self.package_folder, "lib")
+        installincludedir = os.path.join(self.package_folder, "include")
+        additional_options=""
+
+        copy(self, "License.txt", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+
+        copy(self, "*", src=os.path.join(self.source_folder, "OpenHome", "Net", "ServiceGen"), dst=os.path.join(self.package_folder, "lib", "ServiceGen"))
+        mkdir(self, os.path.join(self.package_folder, "lib", "t4"))
+
+        with chdir(self, self.source_folder):
+            if is_msvc(self):
+                if self.settings.build_type == "Debug":
+                    additional_options += " debug=1"
+                self.run(f"nmake /f OhNet.mak install installdir={self.package_folder} installlibdir={installlibdir} installincludedir={installincludedir} {additional_options}")
+            else:
+                args = [f"prefix={self.package_folder}", f"installlibdir={installlibdir}", f"installincludedir={installincludedir}", "rsync=no"]
+                args = self._get_openhome_architecture(args)
+                autotools = Autotools(self)
+                autotools.make(args=args, target="install-libs install-includes")
+                if is_apple_os(self):
+                    fix_apple_shared_install_name(self)
+
+    def package_info(self):
+        self.cpp_info.components["ohNet"].libs = ["ohNet"]
+        self.cpp_info.components["ohNet"].builddirs = [os.path.join("lib", "ServiceGen")]
+        self.cpp_info.components["ohNet"].set_property("cmake_target_name", "ohNet")
+        self.cpp_info.components["ohNet"].defines.extend(["DEFINE_LITTLE_ENDIAN"])
```

Comment:
This seems architecture specific here and like something we shouldn't be defining without respect to the target architecture. Of course, we don't really do anything to define the endianness of architectures which can be big or little endian. Can the upstream not solve this in a more portable way?

Reply:
There are many things I'd wish upstream solved in a more portable way, starting with using CMake instead of Makefiles that do everything including cross-compilation guesses.

There used to be a PR with CMake even, but  don't think any maintainer even looked at it.

That said, from a pragmatic point of view, I prefer to work around such issues on a higher level (Conan package in this instance) and thus have a working solution as opposed to waiting indefinitely for upstream support.

Is there a way in Conan to check and set the endianness in a way that should be compatible accross platforms?

Reply:
I don't think so since the `arch` settings don't really include endianness. Arm can be either big or little endian. It's likely it would require a patch to the source code to calculate it at compile time. Another option is to omit it here and require that consumers define it themselves. You could just define `DEFINE_LITTLE_ENDIAN` in the test package then to have it build on CCI.

Reply:
That's what I did in the first place, but @uilianries directed me at the defines and that they might be a better option. And I kinda agree.

Funny things, whereas upstream deifines `DEFINE_LITTLE_ENDIAN` several times in the Makefiles, `DEFINE_BIG_ENDIAN` is never there, so it seems like they don't even support any Big Endian platforms.